jena-arq/src/main/java/org/apache/jena/sparql/exec/JsonResults.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.jena.sparql.exec;
+
+import java.util.Iterator;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+import org.apache.jena.atlas.json.JsonArray;
+import org.apache.jena.atlas.json.JsonObject;
+import org.apache.jena.atlas.json.JsonValue;
+import org.apache.jena.graph.Node;
+import org.apache.jena.sparql.core.Var;
+import org.apache.jena.sparql.engine.QueryIterator;
+import org.apache.jena.sparql.engine.binding.Binding;
+import org.apache.jena.sparql.lib.RDFTerm2Json;
+
+/**
+ * ARQ supports an extension query type:
+ * <pre>
+ *    JSON {
+ *       "x": ?var1 ,
+ *       "y": ?var2
+ *    } WHERE {
+ *       ...
+ *    }
+ * </pre>
+ */
+public class JsonResults {
```

Comment:
Thank you!! I more or less had an idea of what was missing, but didn't know where to start looking for the right place to fix it.

Reply:
The code is old with a slightly troubled history, and despite being in the new query execution code, it was a copy of the code before the QueryExec reworking. Looking at it now, there was duplication so some DRY applied.